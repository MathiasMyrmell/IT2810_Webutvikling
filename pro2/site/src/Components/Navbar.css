.navbar {
    background-color: rgb(40, 44, 52);
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
    column-gap: 32px;
}

.navbar>div>a {
    color: rgb(137, 227, 252);
    padding: 16px 0;
    text-decoration: none;
    text-align: center;
    display: block;
    font-size: clamp(16px, 2vw, 32px);
}

.navbar>div:hover {
    background-color: rgb(54, 58, 66);
}

/* For mobile phones */
@media (max-width: 512px) {
    .navbar {
        flex-direction: column;
    }

    .navbar>div>a {
        width: 100%;
        padding: 10px 0;
    }
}