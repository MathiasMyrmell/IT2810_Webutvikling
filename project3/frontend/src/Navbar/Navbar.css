.navbar {
    background-color: rgb(34, 34, 34);
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
    column-gap: 32px;
}

.navbar>div>a {
    color: #FFFFFF;
    padding: 16px 10px;
    text-decoration: none;
    text-align: center;
    display: block;
    font-size: clamp(16px, 2vw, 32px);
}

.navbar>div:hover {
    background-color: rgb(20, 20, 20);
}

/* For mobile phones */
@media (max-width: 512px) {
    .navbar {
        flex-direction: column;
    }

    .navbar>div>a {
        width: 100%;
        padding: 10px 0;
    }
}