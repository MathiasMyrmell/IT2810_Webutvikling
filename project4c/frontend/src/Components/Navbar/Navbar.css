.navbar {
    background-color: rgb(34, 34, 34);
    margin: 0;
    padding: 0;
    display: flex;
    flex-direction: row;
    flex-wrap: nowrap;
    justify-content: center;
    column-gap: 32px;
}

.link-styles {
    color: #FFFFFF;
    padding: 16px 10px;
    text-decoration: none;
    text-align: center;
    display: block;
    font-size: clamp(16px, 2vw, 32px);
}

.link-styles:hover {
    background-color: rgb(20, 20, 20);
}

/* For mobile phones */
@media (max-width: 512px) {
    .navbar {
        flex-direction: column;
    }

    .link-styles {
        width: 100%;
        padding: 10px 0;
    }
}