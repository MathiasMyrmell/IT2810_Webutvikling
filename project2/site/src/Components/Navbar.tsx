import React from 'react';
import './Navbar.css';

export default class Navbar extends React.Component {
    render() {
        return <div className="navbar">
            <div><a href="/project2/">Home</a></div>
            <div><a href="/project2/commits">Commits</a></div>
            <div><a href="/project2/issues">Issues</a></div>
        </div>
    }
}