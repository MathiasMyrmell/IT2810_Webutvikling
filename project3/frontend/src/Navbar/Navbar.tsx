import React from 'react';
import './Navbar.css';

export default class Navbar extends React.Component {
    render() {
        return <nav className="navbar">
            <div><a href="/project3/">Recipes</a></div>
            <div><a href="/project3/make%20recipe">Make Recipes</a></div>
        </nav>
    }
}