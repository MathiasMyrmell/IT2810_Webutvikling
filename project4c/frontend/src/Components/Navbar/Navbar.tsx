import React from 'react';
import {Link} from 'react-router-dom';
import "./Navbar.css";

export default function Navbar(){
    
        return (
            <nav className="navbar">
                <Link className='link-styles' to="/project4c/">Recipes</Link>
                <Link className='link-styles' to="/make%20recipe">Make Recipes</Link>
            </nav>
        );
    
}