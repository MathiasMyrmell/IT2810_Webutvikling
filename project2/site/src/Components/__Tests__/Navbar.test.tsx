import renderer from 'react-test-renderer';
import Navbar from '../Navbar';

test('Snapshot test Navbar', () => {
    const tree = renderer.create(<Navbar />).toJSON();
    expect(tree).toMatchSnapshot();
});
