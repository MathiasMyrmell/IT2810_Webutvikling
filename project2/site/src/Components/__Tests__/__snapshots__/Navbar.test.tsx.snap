// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Snapshot test Navbar 1`] = `
<div
  className="navbar"
>
  <div>
    <a
      href="/project2/"
    >
      Home
    </a>
  </div>
  <div>
    <a
      href="/project2/commits"
    >
      Commits
    </a>
  </div>
  <div>
    <a
      href="/project2/issues"
    >
      Issues
    </a>
  </div>
</div>
`;
